drivers/include/tcs37727.h

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (C) 2015 PHYTEC Messtechnik GmbH
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    drivers_tcs37727 TCS37727 Light-To-Digital Converter
+ *
+ * @ingroup     drivers
+ * @brief       Driver for the AMS TCS37727 Color Light-To-Digital Converter
+ *
+ *
+ * @{
+ *
+ * @file
+ * @brief       Interface definition for the TCS37727 sensor driver.
+ *
+ * @author      Felix Siebel <f.siebel@phytec.de>
+ * @author      Johann Fischer <j.fischer@phytec.de>
+ */
+
+#ifndef TCS37727_H
+#define TCS37727_H
+
+#include <stdint.h>
+#include <stdbool.h>
+#include "periph/i2c.h"
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+
+#ifndef TCS37727_I2C_ADDRESS
+#define TCS37727_I2C_ADDRESS           0x29 /**< Default Device Address */
+#endif
+
+/**
+ * @brief Device descriptor for TCS37727 sensors.
+ */
+typedef struct {
+    i2c_t i2c;              /**< I2C device the sensor is connected to */
+    uint8_t addr;           /**< the sensor's slave address on the I2C bus */
+    bool initialized;       /**< sensor status, true if sensor is initialized */
+    int atime_us;           /**< atime value conveted to microseconds */
+    int again;              /**< amount of gain */
+    int ir;                 /**< IR component */
+    int cr;                 /**< IR compensated channels red */
+    int cg;                 /**< IR compensated channels green */
+    int cb;                 /**< IR compensated channels blue */
+    int cc;                 /**< IR compensated channels clear */
+    int lux;                /**< Lux */
+    int ct;                 /**< Color temperature */
+} tcs37727_t;
```

Comment:
once again:
The init function do not need any parameters, the sensor should be self adjusted by autgain function.
I am considering to divide the tcs37727_t in tcs37727_t and tcs37727_data_t, there also insert color_rgb_t. What do you mean?